src/goto-symex/symex_goto.cpp

```diff
@@ -220,7 +220,17 @@ void goto_symext::symex_goto(statet &state)
   // merge_gotos when we visit new_state_pc
   framet::goto_state_listt &goto_state_list =
     state.top().goto_state_map[new_state_pc];
-  goto_state_list.emplace_back(state);
+
+  // If the guard is true (and the alternative branch unreachable) we can move
+  // the state in this case as it'll never be accessed on the alternate branch.
+  if(new_guard.is_true())
+  {
+    goto_state_list.emplace_back(std::move(state));
+  }
+  else
+  {
+    goto_state_list.emplace_back(state);
+  }
```

Comment:
We need to find a better way for doing this: we now have, with only a `symex_transition` in between, the same `if(new_guard.is_true())` branching structure. This is hard to reason about for compilers and humans alike.